Simplify map generator Player and Symmetry options

pchote opened the pull request:
Player now specifies the total player count, and the Symmetry options (which combine the previous Mirror and Rotations) adjust to show only valid options. This makes it easier to generate valid maps with the expected number of players.

~~Depends on #21849.~~

pchote committed:
Remove derived properties from Parameters.

pchote commented:
Rebased.

PunkPun commented on mods/cnc/fluent/rules.ftl:
```diff
@@ -821,19 +821,31 @@ label-cnc-map-generator-choice-terrain-type-mountain-lakes =
+label-cnc-map-generator-choice-symmetry-mirror-vertical =
+   .label = Mirror Vertical
+label-cnc-map-generator-choice-symmetry-mirror-diagonal-tl =
+   .label = Mirror Diagonal (TL-BR)
```
```suggestion
   .label = Mirror Diagonal (Top-Left)
```

The naming doesn't feel right. I know we use it internally but it's never exposed to users.

Maybe we could just mention one corner of the mirror? The other option being Top-Right?

PunkPun reviewed (commented):
Overall it's feels like a good usability improvement

It's a bit weird to limit to 8 rotations though, when we can easilly handle more for large maps.

pchote commented:
> It's a bit weird to limit to 8 rotations though, when we can easilly handle more for large maps.

How strongly do you feel about this point? We could in principle add also a Size filter for specific options, but i imagine that will invite some bike-shedding over where we set the limits.

pchote commented:
(or, equally, apply a player limit on the skirmish map size option)

PunkPun commented:
More rotations could be added back later on along side player limits

pchote committed:
Simplify Player and Symmetry options.

Player now specifies the total player count,
and the Symmetry options adjust to show only
valid options.

pchote commented:
Updated names.

PunkPun reviewed (approved):
ok

PunkPun commented:
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/eb2a99f40b29e4044519ea40f951818537117d6f...844514b8a35b985a64e0ab5aea5a4d5b1bd709e8)
